src/components/portrait/portrait.pw.tsx

```diff
@@ -0,0 +1,486 @@
+    await mount(
+      <PortraitDefaultComponent src={testImage} alt="playwright-test" />
+    );
+
+    await expect(portraitImage(page)).toHaveAttribute("alt", "playwright-test");
+  });
+
+  [
+    ["square", "0px"],
+    ["circle", "50%"],
+  ].forEach(([shape, radius]) => {
+    test(`should render with shape prop passed as ${shape}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PortraitDefaultComponent shape={shape} />);
+
+      await expect(portraitPreview(page).locator("span")).toHaveCSS(
+        "border-radius",
+        radius
+      );
+    });
+  });
+
+  ["error", "warning", "info"].forEach((iconType) => {
+    test(`should render with iconType prop passed as ${iconType}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PortraitDefaultComponent iconType={iconType} />);
+
+      await expect(icon(page)).toBeVisible();
+      await expect(icon(page)).toHaveAttribute("data-element", iconType);
+    });
+  });
+
+  ["SPM", "JM", "AR", "MJ"].forEach((passInitials) => {
+    test(`should render with correct width and height, when initials prop is passed as ${passInitials}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PortraitDefaultComponent initials={passInitials} />);
+
+      await expect(portraitInitials(page)).toHaveCSS("height", "38px");
+      await expect(portraitInitials(page)).toHaveCSS("width", "38px");
+    });
+  });
+
+  ([
+    ["without", false, "rgb(242, 245, 246)"],
+    ["with", true, "rgb(153, 173, 183)"],
+  ] as [string, boolean, string][]).forEach(([renderState, boolVal, color]) => {
+    test(`should render ${renderState} dark background variant, when darkBackground prop is ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PortraitDefaultComponent darkBackground={boolVal} />);
+
+      await expect(portraitPreview(page).locator("span")).toHaveCSS(
+        "background-color",
+        color
+      );
+    });
+  });
+
+  testData.forEach((tooltipMessage) => {
+    test(`should render with toolTipMessage as ${tooltipMessage}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PortraitComponent tooltipMessage={tooltipMessage} />);
+
+      await portraitPreview(page).hover();
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveText(tooltipMessage);
+    });
+  });
+
+  testData.forEach((tooltipId) => {
+    test(`should render with tooltipId as ${tooltipId}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <PortraitComponent tooltipMessage="foo" tooltipId={tooltipId} />
+      );
+
+      await portraitPreview(page).hover();
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveId(tooltipId);
+    });
+  });
+
+  visibilityValues.forEach(([boolVal, renderState]) => {
```

Comment:
**comment** (non-blocking): I see you've combined what was 2 separate Cypress tests here into one with a `forEach` to cover both cases. I'm not sure this is an improvement - there wasn't a great deal of repetition between the two tests, and the tests are in fact different (as we only check the text in the visible case). Personally I prefer them separate, but again I'm happy to leave up to you.

Reply:
Yeah fair point, I personally prefer using a `forEach` here as I know technically we're testing different things (being visible and not being visible), but I feel as though there is enough overlap elsewhere to justify it